backend/src/sockets/actions.ts

```diff
@@ -0,0 +1,38 @@
+import { getNumberOfUsersInAllRooms, getNumberOfUsersInRoom } from 'shared/helpers/sockets';
+import { io } from './index';
+import { ISocket } from 'shared/sockets/types';
+
+class SocketActions {
+  onConnect = () => {
+    console.log('A user connected');
+  };
+
+  onDisconnect = () => {
+    console.log('A user disconnected');
+  };
+
+  onJoinRoom = (socket: ISocket, roomId: string) => {
+    socket.join(roomId);
+    console.log(`${socket.username} joined room ${roomId} with ${getNumberOfUsersInRoom(roomId)} active users`);
+  };
+
+  onExitRoom = (socket: ISocket, roomId: string) => {
+    socket.leave(roomId);
+    console.log(`${socket.username} left room ${roomId}`);
+  };
+
+  emitNumberOfUsersInAllRooms = (socket?: ISocket) => {
+    if (!socket) {
+      io.emit('total-number-of-users', getNumberOfUsersInAllRooms());
+      return;
+    }
+
+    socket.emit('total-number-of-users', getNumberOfUsersInAllRooms());
```

Comment:
```suggestion
const emitter = socket || io;
emitter.emit('total-number-of-users', getNumberOfUsersInAllRooms());
```
I do not know if this will cause any `TypeScript` issues.

Reply:
I was sure something like that didn't even work in `TS` but apparently it does! Love it! ❤️